STYLEGUIDE.md

```diff
@@ -0,0 +1,175 @@
+# Style guide and design principles
+
+This guide provides a set of design principles and guidelines for the `yoneda`
+project. Our style and design principles take heavy inspiration from
+[`agda-unimath`](https://github.com/UniMath/agda-unimath).
+
+## The structure of code
+
+We enforce strict formatting rules. This formatting allows the type of the
+defined term to be easily readable, and aids in understanding the structure of
+the definition.
+
+The general format of a definition is as follows:
+
+```rzk
+#def concat
+  ( p : x = y)
```

Comment:
Why is there a space after `(` but not before `)`?

Reply:
This is the Lisp-like style convention of agda-unimath. The space directly after `(` has a very clear purpose (to align code with indentation levels. A space before `)` does not have a purpose. See for instance how the parentheses are used in the other example.

Reply:
Note that in general the convention should only stipulate having a space after `(` when it is only preceded by other opening parentheses on that line. However, I also do this for after Sigmas and other 1-character symbols

Reply:
I guess I am not used to this imbalance, and it kind of annoys me that treatment of parentheses is not symmetric :) can't we solve the problem by saying that indentation is 3 spaces (instead of 2) and parenthesis is a part of the indentation (punctuation is dimmed in the editor anyway)?

That said, I am not strongly against, this is a very minor point, I can get used to it :)

Reply:
I don't have strong feelings about parentheses but I do like the space after lambdas \ --- can we mention this explicitly somewhere? 

The reason is in vsCode I often run a command to convert LaTeX symbols to unicode, but it does funny things to some of my lambda terms eg "\k".